### PR TITLE
Feature/post details

### LINKED ISSUE #10: As a user, I should be able to click on a post and see the url change

### LINKED ISSUE #37: As a user, when I click on a post, I should see the description underneath images (if images)

### LINKED ISSUE #38: As a user when I click on a post, I should see a section for comments

### PR DESCRIPTION
**Pull request by clairefields15**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Linter has passed locally and any fixes were made for failures
- [ ] CircleCI build is passing
- [ ] All tests passing locally

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## What is the current behavior?
There is no post details page

## What is the new behavior?
You can click on a post preview, see the URL change, and are directed to a new page with details.
Styling is very much a WIP, I mainly just added grid, borders and created all the necessary components so that we could then divide and conquer the rest of it easily.
I also did some general refactoring: created a utils folder that has a file for all types that are used throughout the application (posts, users, comments, things like that), and moved the date formatting functionality into a util file so that it could be used by multiple components. Also broke out the div for a user's avatar, name, and post time into a component so that it could be reused.

- closes #37 
- closes #38 
- closes #10

### CODE REVIEW

File: src/components/PostDetails/PostDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }, [data, loading]);
+
+  return (
+    <>
+      {error && <p>{error.message}</p>}
+      {loading && <p>Loading post...</p>}
+      {details && !loading && !error && (
+        <main className='post-details-section'>
+          <section className='post-content'>
+            <h2>Title: {details.title}</h2>
```
Comment (k-atwhite) [inline src/components/PostDetails/PostDetails.tsx:31]: Small change, but I think we don't need to say this is a "title". I think the styling will speak for iteself

File: src/index.tsx [hunk 2]
```diff
@@ -34,6 +35,37 @@ export const GET_ALL_POSTS = gql`
   }
 `;
 
+export const GET_POST_DETAILS = gql`
```
Comment (k-atwhite) [inline src/index.tsx:38]: Does the gql call need to live in index.tsx?

**Review by k-atwhite — APPROVED** [review]
I had a couple tiny suggestions. But overall this looks AMAZING. Nice job reusing "Posted By"

File: src/index.tsx (hunk 2 above)
Comment (clairefields15) [reply src/index.tsx:38]: It is best practice for it to live there, yeah

**Pull request merged**